timemachine/potentials/nonbonded.py

```diff
@@ -25,6 +20,45 @@ def nonbonded_v3(
     cutoff,
     lambda_plane_idxs,
     lambda_offset_idxs):
+    """Lennard-Jones + Coulomb, with a few important twists:
+    * distances are computed in 4D, controlled by lambda, lambda_plane_idxs, lambda_offset_idxs
+    * each pairwise LJ and Coulomb term can be multiplied by an adjustable rescale_mask parameter
+    * Coulomb terms are multiplied by erfc(beta * distance)
+
+    Parameters
+    ----------
+    conf : (N, 3) or (N, 4) np.array
+        3D or 4D coordinates
+        if 3D, will be converted to 4D using (x,y,z) -> (x,y,z,w)
+            where w = cutoff * (lambda_plane_idxs + lambda_offset_idxs * lamb)
+    params : (N, 3) np.array
+        columns [charges, sigmas, epsilons], one row per particle
+    box : Optional 3x3 np.array
+    lamb : float
+    charge_rescale_mask : (N, N) np.array
+        the Coulomb contribution of pair (i,j) will be multiplied by charge_rescale_mask[i,j]
+    lj_rescale_mask : (N, N) np.array
+        the Lennard-Jones contribution of pair (i,j) will be multiplied by lj_rescale_mask[i,j]
+    scales
+        unused # TODO: remove?
+    beta : float
+        the charge product q_ij will be multiplied by erfc(beta*d_ij)
+    cutoff : Optional float
```

Comment:
at some point we should add a sanity check that U at cutoff is less than some threshold